pkg/registry/populator.go

```diff
@@ -16,6 +17,10 @@ import (
 	"github.com/operator-framework/operator-registry/pkg/image"
 )
 
+type Dependencies struct {
+	DependenciesStr []map[string]string `json:"dependencies" yaml:"dependencies"`
```

Comment:
I fixed the naming. 
Given the dependency syntax is a list of fields (not nested):
```
dependencies:
  - type: olm.package
    packgeName: prometheus
    version: >0.27.0
  - type: olm.gvk
    group: etcd.database.coreos.com
    kind: EtcdCluster
    version: v1beta2
```
I use `Dependencies` struct to get the JSON string of each item (which will be injected into `Value` in `DependenciesFile`) while using `DependenciesFile` to get the `Type`. This is to avoid using nested structure in the syntax. We already agree on the `Dependency` type only contains `Type` and `Value`. When client/user gets the string from `Value`, they can unmarshal into `GVKDependency` or `PackageDependency` type that I specify in `types.go` in `registry` package.